src/Blumchen/Database/Run.cs

```diff
@@ -16,11 +13,11 @@ public static class Run
         CancellationToken ct)
     {
         await using var command = dataSource.CreateCommand(sql);
-        await command.ExecuteNonQueryAsync(ct).ConfigureAwait(false);
+        await command.ExecuteNonQueryAsync(ct);
     }
 
     public static async Task EnsureTableExists(this NpgsqlDataSource dataSource, TableDescriptorBuilder.MessageTable tableDescriptor, CancellationToken ct)
-        => await dataSource.Execute(tableDescriptor.ToString(), ct).ConfigureAwait(false);
+        => await dataSource.Execute(string.Concat("select pg_advisory_xact_lock(12345);", tableDescriptor), ct).ConfigureAwait(false);
```

Comment:
**QUESTION:** Is the advisory lock used here to ensure that we're not setting up the table multiple times at the same time?

Reply:
Exactly. It happens always when dealing with multiple ServiceWorker instances